### PR TITLE
(doc) Add deprecated CRD version procedure and note

### PR DESCRIPTION
**Pull request by dinhxuanvu**
Signed-off-by: Vu Dinh <vdinh@redhat.com>

### CODE REVIEW

File: Documentation/design/dependency-resolution.md [hunk 1]
```diff
@@ -16,8 +16,83 @@ The following examples motivate why OLM's dependency resolution and upgrade stra
...
-- All existing versions in current CRD are present in new CRD
-- All existing instances (Custom Resource) of current CRD are valid when validated against new CRD's validation schema
+- All existing serving versions in current CRD are present in new CRD
+- All existing instances (Custom Resource) that are associated with serving versions of CRD are valid when validated against new CRD's validation schema
+
+### Add a new version to CRD
+
+The recommended procedure to add a new version in CRD:
+
+1. For example, the current CRD has one version `v1alpha1` and you want to a new version `v1beta1` and mark it as the new storage version:
```
Comment (jpeeler) [inline Documentation/design/dependency-resolution.md]: want to move to a

File: Documentation/design/dependency-resolution.md [hunk 2]
```diff
@@ -16,8 +16,83 @@ The following examples motivate why OLM's dependency resolution and upgrade stra
...
+versions:
+  - name: v1beta1
+    served: true
+    storage: true
+```
+
+Note:
+
+1. In order to remove a specific version that is or was storage version from CRD, that version needs to be removed from
+`storedVersion` in CRD's status. OLM will attempt to do this for you if it detects a stored version is no longer existed in new CRD.
```
Comment (jpeeler) [inline Documentation/design/dependency-resolution.md]: ~is~ no longer exists in the new CRD.

**Comment by dinhxuanvu** [comment]
Need those PRs to be merged first #973 and #983

**Pull request merged**